### PR TITLE
[ci skip] Enhanced Rdoc for Hash

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Changes for methods:
* #keep_if
* #clear
* #[]= and #store
* #replace
* #length and #size 
* #empty?

### CODE REVIEW

File: hash.c [hunk 1]
```diff
@@ -3108,29 +3120,36 @@ NOINSERT_UPDATE_CALLBACK(hash_aset_str)
...
- *     h.key(100).equal? a #=> false
- *     h.key(200).equal? b #=> true
+ *  ---
  *
+ *  Raises an exception if +key+ is invalid
+ *  (see {Invalid Hash Keys}[#class-Hash-label-Invalid+Hash+Keys]):
+ *    h = {foo: 0, bar: 1}
+ *    # Raises NoMethodError (undefined method `hash' for #<BasicObject>)
+ *    h[BasicObject.new] = 2
+ *    # Raises NoMethodError (undefined method `hash' for #<BasicObject>)
```
Comment (ioquatix) [inline hash.c:3151]: Nit, add trailing `:`? above too.

File: hash.c [hunk 2]
```diff
@@ -3234,12 +3256,11 @@ rb_hash_size_num(VALUE hash)
 
 /*
  *  call-seq:
- *     hsh.empty?    -> true or false
- *
- *  Returns <code>true</code> if <i>hsh</i> contains no key-value pairs.
- *
- *     {}.empty?   #=> true
+ *    hash.empty? → true or false
```
Comment (ioquatix) [inline hash.c:3259]: This arrow appears to be unicode, is that okay? In other places it was `->`.

**Comment by ioquatix** [comment]
There are a few minor points you might like to address. Great work.

File: hash.c (hunk 1 above)
Comment (BurdetteLamar) [reply hash.c]: Done.

File: hash.c (hunk 2 above)
Comment (BurdetteLamar) [reply hash.c]: Fixed.

**Comment by BurdetteLamar** [comment]
Thanks, @ioquatix, for your careful review.

**Comment by BurdetteLamar** [comment]
This is doc-only, so should not have a problem with failed or cancelled tests.

**Comment by ioquatix** [comment]
When you merge this, please use "Squash and merge" otherwise we end up with 6+ commits for something that should really be one. Please reformat the commit message appropriately.

**Pull request merged**